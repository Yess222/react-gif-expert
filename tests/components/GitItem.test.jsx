import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GitItem />' , () => {

    const title = 'Un titulo'
    const url = 'https://localhost/algo.jpg'

    test('debe de mostrar el componente correctamente', () => {
        
        const {container} = render( <GifItem title={ title } url={ url } />)
        expect(container).toMatchSnapshot();

    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render( <GifItem title={ title } url={ url } />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('debe de mostrar el titulo en el componente', () => {
        
    })

})