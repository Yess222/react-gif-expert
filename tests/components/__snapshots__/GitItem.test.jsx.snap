// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem /> debe de mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
       
    </p>
  </div>
</div>
`;
